server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java

```diff
@@ -77,12 +82,25 @@ protected Logger getLogger() {
     public void configure(HierarchicalConfiguration config) throws ConfigurationException {
         configureDefaultDomain(config);
 
+        addEnvDomain();
+
         setAutoDetect(config.getBoolean(CONFIGURE_AUTODETECT, true));
         setAutoDetectIP(config.getBoolean(CONFIGURE_AUTODETECT_IP, true));
     }
 
+    private void addEnvDomain() {
+        String envDomain = envDetector.getEnv(ENV_DOMAIN);
+        if (!Strings.isNullOrEmpty(envDomain)) {
+            try {
+                LOGGER.info("Adding ENV defined domain {}", envDomain);
```

Comment:
I think you should explain somewhere what ENV is meaning. I understand it while reading this PR, but out of context this can be hard.